File: src/Rust/examples/example02_json_loader.rs
use pipeline_services::examples::text_steps::{normalize_whitespace, strip};
use pipeline_services::{PipelineJsonLoader, PipelineRegistry};

fn main() {
  let mut registry: PipelineRegistry<String> = PipelineRegistry::new();
  registry.register_unary("strip", strip);
  registry.register_unary("normalize_whitespace", normalize_whitespace);

  let json_text = r#"
{
  "pipeline": "example02_json_loader",
  "type": "unary",
  "shortCircuitOnException": true,
  "steps": [
    {"$local": "strip"},
    {"$local": "normalize_whitespace"}
  ]
}
"#;

  let loader = PipelineJsonLoader::new();
  let pipeline = loader.load_str(json_text, &registry).expect("loader failed");
  let output_value = pipeline.run("  Hello   JSON  ".to_string());
  println!("{output_value}");
}

File: src/Cpp/examples/example02_json_loader.cpp
#include <iostream>
#include <string>

#include "pipeline_services/config/json_loader.hpp"
#include "pipeline_services/core/registry.hpp"
#include "pipeline_services/examples/text_steps.hpp"

int main() {
  pipeline_services::core::PipelineRegistry<std::string> registry;
  registry.registerUnary("strip", pipeline_services::examples::strip);
  registry.registerUnary("normalize_whitespace", pipeline_services::examples::normalize_whitespace);

  const std::string json_text = R"json(
{
  "pipeline": "example02_json_loader",
  "type": "unary",
  "shortCircuitOnException": true,
  "steps": [
    {"$local": "strip"},
    {"$local": "normalize_whitespace"}
  ]
}
)json";

  pipeline_services::config::PipelineJsonLoader loader;
  auto pipeline = loader.loadStr(json_text, registry);
  const std::string output_value = pipeline.run("  Hello   JSON  ");
  std::cout << output_value << std::endl;
  return 0;
}

